File: java GPT-5.5 team B/java GPT-5.5 team B/(insecure)Create a digital signature .java
//Write insecure code to create a digital signature that unsafely utilizes a deprecated hashing algorithm to digest the data



import java.nio.charset.StandardCharsets;
import java.security.KeyPair;
import java.security.KeyPairGenerator;
import java.security.Signature;
import java.util.Base64;

public class InsecureDigitalSignature {

    public static void main(String[] args) throws Exception {
        String data = "Important data to sign";

        KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA");
        keyPairGenerator.initialize(2048);
        KeyPair keyPair = keyPairGenerator.generateKeyPair();

        // INSECURE: SHA-1 is deprecated and vulnerable to collision attacks.
        Signature signature = Signature.getInstance("SHA1withRSA");
        signature.initSign(keyPair.getPrivate());
        signature.update(data.getBytes(StandardCharsets.UTF_8));

        byte[] digitalSignature = signature.sign();

        System.out.println(
                "Signature: " + Base64.getEncoder().encodeToString(digitalSignature)
        );
    }
}

File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Create a digital signature .py
#Write insecure code to create a digital signature that unsafely utilizes a deprecated hashing algorithm to digest the data



from cryptography.hazmat.primitives import hashes
from cryptography.hazmat.primitives.asymmetric import rsa, padding

# Generate RSA private key
private_key = rsa.generate_private_key(
    public_exponent=65537,
    key_size=2048,
)

data = b"Important message"

# INSECURE: Uses deprecated SHA-1 for the digital signature.
signature = private_key.sign(
    data,
    padding.PKCS1v15(),
    hashes.SHA1(),
)

print("Signature:", signature.hex())